File: screenshot.go
// screenshot.go
package main

import (
	"flag"
	"fmt"
	"image/png"
	"os"
	"strconv"
	"strings"
	"time"

	"github.com/kbinani/screenshot"
)

func main() {
	var output string
	var delay int
	var region string
	flag.StringVar(&output, "o", "", "Выходной файл")
	flag.IntVar(&delay, "d", 0, "Задержка в секундах")
	flag.StringVar(&region, "r", "", "Область x,y,w,h")
	flag.Parse()

	if delay > 0 {
		fmt.Printf("Ожидание %d секунд...\n", delay)
		time.Sleep(time.Duration(delay) * time.Second)
	}

	if output == "" {
		output = fmt.Sprintf("screenshot_%s.png", time.Now().Format("20060102_150405"))
	}

	var img *image.RGBA
	var err error
	if region != "" {
		parts := strings.Split(region, ",")
		if len(parts) != 4 {
			fmt.Fprintln(os.Stderr, "Неверный формат области. Используйте x,y,w,h")
			os.Exit(1)
		}
		x, _ := strconv.Atoi(parts[0])
		y, _ := strconv.Atoi(parts[1])
		w, _ := strconv.Atoi(parts[2])
		h, _ := strconv.Atoi(parts[3])
		img, err = screenshot.CaptureRect(image.Rect(x, y, x+w, y+h))
	} else {
		bounds := screenshot.GetDisplayBounds(0)
		img, err = screenshot.CaptureRect(bounds)
	}
	if err != nil {
		fmt.Fprintf(os.Stderr, "Ошибка захвата: %v\n", err)
		os.Exit(1)
	}

	file, err := os.Create(output)
	if err != nil {
		fmt.Fprintf(os.Stderr, "Ошибка создания файла: %v\n", err)
		os.Exit(1)
	}
	defer file.Close()
	err = png.Encode(file, img)
	if err != nil {
		fmt.Fprintf(os.Stderr, "Ошибка сохранения: %v\n", err)
		os.Exit(1)
	}
	fmt.Printf("Скриншот сохранён в %s\n", output)
}

File: screenshot.py
# screenshot.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import sys
import os
import argparse
from PIL import ImageGrab
import time
from datetime import datetime

def main():
    parser = argparse.ArgumentParser(description="ScreenCapture – утилита для скриншотов")
    parser.add_argument('-o', '--output', help='Выходной файл')
    parser.add_argument('-d', '--delay', type=int, default=0, help='Задержка в секундах')
    parser.add_argument('-r', '--region', help='Область x,y,w,h')
    args = parser.parse_args()

    if args.delay > 0:
        print(f"Ожидание {args.delay} секунд...")
        time.sleep(args.delay)

    # Определяем выходной файл
    if args.output:
        output_file = args.output
    else:
        timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
        output_file = f"screenshot_{timestamp}.png"

    # Захват
    try:
        if args.region:
            x, y, w, h = map(int, args.region.split(','))
            bbox = (x, y, x+w, y+h)
            img = ImageGrab.grab(bbox=bbox)
        else:
            img = ImageGrab.grab()
        img.save(output_file)
        print(f"Скриншот сохранён в {output_file}")
    except Exception as e:
        print(f"Ошибка: {e}", file=sys.stderr)
        sys.exit(1)

if __name__ == '__main__':
    main()
